BLL/ViewModels/ClientViewModel.cs

```diff
@@ -0,0 +1,12 @@
+﻿using DAL.Entities;
+
+namespace BLL.ViewModels;
+
+public record ClientViewModel(
+    Guid Id,
+    string FullName,
+    string Email,
+    string PhoneNumber,
+    Vehicle? Vehicle,
+    IEnumerable<VehicleClientHistory> VehicleClientHistories
+    );
```

Comment:
ViewModels should be in API layer.
API - ViewModels
BLL - Models
DAL - Entities
In your case "Models" are redundant